scico/denoiser.py

```diff
@@ -90,6 +100,63 @@ def bm3d(x: JaxArray, sigma: float, is_rgb: bool = False):
     return y
 
 
+def bm4d(x: JaxArray, sigma: float):
+    r"""An interface to the BM4D denoiser :cite:`maggioni-2012-nonlocal`.
+
+    BM4D denoising is performed using the
+    `code <https://pypi.org/project/bm4d/>`__ released by the authors of
+    :cite:`maggioni-2012-nonlocal`.
+
+    Args:
+        x: Input image. Expected to be a 3D array. Higher dimensional
+            arrays are tolerated only if the additional dimensions are
+            singletons.
+        sigma: Noise parameter.
+
+    Returns:
+        Denoised output.
+    """
+    if not have_bm4d:
+        raise RuntimeError("Package bm4d is required for use of this function.")
+
+    bm4d_eval = tunibm4d.bm4d
+
+    if np.iscomplexobj(x):
+        raise TypeError(f"BM4D requires real-valued inputs, got {x.dtype}")
+
+    # Support arrays with more than three axes when the additional axes are singletons.
+    x_in_shape = x.shape
+
+    if isinstance(x.ndim, tuple) or x.ndim < 3:
+        raise ValueError(f"BM4D requires three dimensional inputs; got ndim = {x.ndim}")
+
+    # This check is also performed inside the BM4D call, but due to the host_callback,
+    # no exception is raised and the program will crash with no traceback.
+    # NOTE: if BM4D is extended to allow for different profiles, the block size must be
+    #       updated; this presumes 'np' profile (bs=8)
```

Comment:
Have you confirmed that the comment on profiles also applies to BM4D, or is it possibly just residual from the initial copy from the BM4D code?

Reply:
Yes, it applies to bm4d as well.